packages/viewer/src/rendering/canvas.ts

```diff
@@ -134,3 +138,21 @@ export function createCanvasRenderer(): CanvasRenderer {
     return data.frame;
   };
 }
+
+export function createCanvasDepthProvider(
+  imageScaleProvider: ImageScaleProvider,
+  canvas?: CanvasRenderingContext2D | null
+): CanvasDepthProvider {
+  return (point) => {
+    if (canvas != null) {
+      const scale = imageScaleProvider();
+      const scaled = Point.scale(point, scale?.x || 1, scale?.y || 1);
+      const data = canvas
+        .getImageData(scaled.x, scaled.y, 1, 1)
+        .data.slice(0, 1);
+
+      return data[0] / 255.0;
```

Comment:
Depth value calculation. `data[0]` will be `0` at the near clipping plane and `255` at the far clipping plane